src/System.Text.Json.Dynamic/System/Text/Json/JsonDynamicObject.cs

```diff
@@ -25,54 +25,54 @@ public static JsonDynamicObject Parse(ReadOnlySpan<byte> utf8, int expectedNumbe
             var properties = new Dictionary<JsonProperty, JsonValue>(expectedNumberOfProperties);
             stack.Push(new JsonDynamicObject(properties));
 
-            var reader = new JsonReader(new Utf8String(utf8));
+            var reader = new JsonReader(utf8, TextEncoder.Utf8);
             while (reader.Read())
             {
                 switch (reader.TokenType)
                 {
-                    case JsonReader.JsonTokenType.Property:
-                        var name = reader.GetName();
-                        var type = reader.GetJsonValueType();
-                        var value = reader.GetValue();
+                    case JsonTokenType.PropertyName:
+                        var name = new Utf8String(reader.Value);
```

Comment:
I think we should avoid taking dependencies on Utf8String. I worry that we will ship the JSON APIs before we finish the design of Utf8String.